tests/unit/challenges/test_views.py

```diff
@@ -3679,3 +3679,127 @@ def test_star_challenge(self):
         response = self.client.post(self.url, {})
         self.assertEqual(response.data, expected)
         self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+
+class GetChallengePhaseByPk(BaseChallengePhaseClass):
+
+    def setUp(self):
+        super(GetChallengePhaseByPk, self).setUp()
+        self.url = reverse_lazy('challenges:get_challenge_phase_by_pk',
+                                kwargs={'pk': self.challenge_phase.pk})
+
+    def test_get_challenge_phase_by_pk(self):
+        expected = {
+            "id": self.challenge_phase.id,
+            "name": self.challenge_phase.name,
+            "description": self.challenge_phase.description,
+            "leaderboard_public": self.challenge_phase.leaderboard_public,
+            "start_date": "{0}{1}".format(self.challenge_phase.start_date.isoformat(), 'Z').replace("+00:00", ""),
+            "end_date": "{0}{1}".format(self.challenge_phase.end_date.isoformat(), 'Z').replace("+00:00", ""),
+            "challenge": self.challenge_phase.challenge.pk,
+            "max_submissions_per_day": self.challenge_phase.max_submissions_per_day,
+            'max_submissions_per_month': self.challenge_phase.max_submissions_per_month,
+            "max_submissions": self.challenge_phase.max_submissions,
+            "is_public": self.challenge_phase.is_public,
+            "is_active": True,
+            "codename": self.challenge_phase.codename,
+            "slug": self.challenge_phase.slug,
+        }
+        response = self.client.get(self.url, {})
+        self.assertEqual(response.data, expected)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_get_challenge_phase_by_pk_does_not_exist(self):
+        self.url = reverse_lazy('challenges:get_challenge_phase_by_pk',
+                                kwargs={'pk': self.challenge_phase.pk + 2})
+        expected = {
+            'detail': 'ChallengePhase {} does not exist'.format(self.challenge_phase.pk + 2)
+        }
+        response = self.client.get(self.url, {})
+        self.assertEqual(response.data, expected)
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+
+class GetChallengePhasesByChallengePk(BaseChallengePhaseClass):
+
+    def setUp(self):
+        super(GetChallengePhasesByChallengePk, self).setUp()
+        self.url = reverse_lazy('challenges:get_challenge_phases_by_challenge_pk',
+                                kwargs={"challenge_pk": self.challenge.pk},)
+
+        self.user1 = User.objects.create(
+            username="someuser1",
+            email="user1@test.com",
+            password="secret_psassword",
+        )
+
+        EmailAddress.objects.create(
+            user=self.user1,
+            email="user1@test.com",
+            primary=True,
+            verified=True,
+        )
+
```

Comment:
One test case is missing when the `challenge_pk` is invalid. Can you please add that.